ynr/apps/moderation_queue/tests/test_changes_for_review.py

```diff
@@ -257,6 +258,85 @@ def test_users_with_past_approved_edits_dont_need_review(self):
             PREVIOUSLY_APPROVED_COUNT + 1,
         )
 
+    def update_person_and_version(self, user, example_person):
+        response = self.app.get(
+            "/person/{person_id}/update".format(person_id=example_person.id),
+            user=self.user,
+        )
+
+        form_data = response.forms["person-details"]
+        form_data["favourite_biscuit"] = "Jammy Dodgers"
+        form_data.submit()
+
+        LoggedAction.objects.create(
+            user=self.user,
+            action_type=ActionType.PERSON_UPDATE,
+            person=example_person,
+            popit_person_new_version=random_person_id(),
+            source="Just for tests...",
+        )
+        example_person.refresh_from_db()
+        example_person.favourite_biscuit = "Jammy Dodgers"
+        example_person.record_version(get_change_metadata(None, "Biscuit edit"))
+        example_person.save()
+        return example_person
+
+    def revert_edit_to_person(self, user, example_person):
+        first_version = example_person.versions[1]
+        # I keep hitting a candidacy error so skipping this approach for now
+        # revert_person_from_version_data(example_person, last_version["data"])
+        # Instead...
+
+        example_person.favourite_biscuit = first_version["data"][
+            "extra_fields"
+        ]["favourite_biscuits"]
+        example_person.save()
+
+        LoggedAction.objects.create(
+            user=self.user,
+            action_type=ActionType.PERSON_REVERT,
+            person=example_person,
+            popit_person_new_version=random_person_id(),
+            source="Revert to previous version",
+        )
+
+    def test_reverted_person_edit_needs_review(self, change_metadata=True):
+        """Test that reverted edits are flagged for review"""
+        # create a person
+        example_person = people.tests.factories.PersonFactory.create(
+            id=4758, name="Phil Hutty", favourite_biscuit="Gingerbread"
+        )
+        # create a candidacy for reverting later
+        example_person.memberships.create(
+            ballot=self.local_ballot,
+            party=self.green_party,
+            post=self.local_ballot.post,
+        )
+        # Save first version of the person
+        example_person.record_version(get_change_metadata(None, "No change"))
+        example_person.save()
+        version = example_person.versions[0]
+        user = self.user
+
+        # update person and record a new version
+        self.update_person_and_version(self.user, example_person)
+
+        # revert to previous version
+        self.revert_edit_to_person(self.user, example_person)
```

Comment:
I think when it comes to reverting the edits, we should go through the [RevertPersonView](https://github.com/DemocracyClub/yournextrepresentative/blob/7a1a61ed7db11965c5fc6ab6973fe749c874b300/ynr/apps/candidates/views/people.py#L472) that does this, in a similar way that the initial update is made via the UpdatePersonView

Reply:
Just realised the link on here is incorrect - meant to be to https://github.com/DemocracyClub/yournextrepresentative/blob/7a1a61ed7db11965c5fc6ab6973fe749c874b300/ynr/apps/candidates/views/people.py#L150